CMakeLists.txt

```diff
@@ -63,6 +63,7 @@ if(COMPILE_WASM)
   list(APPEND WASM_COMPILE_FLAGS -pthread -O3 -g2 -fPIC -mssse3 -msimd128)
   list(APPEND WASM_COMPILE_FLAGS "SHELL:-s WASM=1" "SHELL:-s ASSERTIONS=0" "SHELL:-s DISABLE_EXCEPTION_CATCHING=1" "SHELL:-s LLD_REPORT_UNDEFINED" "SHELL:-s FORCE_FILESYSTEM=1" "SHELL:-s ALLOW_MEMORY_GROWTH=1")
   list(APPEND WASM_COMPILE_FLAGS -Wno-error=pthreads-mem-growth)
+  list(APPEND WASM_COMPILE_FLAGS -DWASM_HIDE_THREADS)
```

Comment:
Didn't we already have a wasm flag? Can we reuse it?

Reply:
There was `WITH_PTHREADS` which was being used, @kpu had complaints.

Reply:
Additional background: sane platform needed to be default (not WASM). And `WITHOUT_PTHREADS` had some issue with the meaning conveyed. I just named the new thing `WASM_HIDE_THREADS` which I think is what I am doing. 

Did I understand the review comment wrong?

Reply:
I was saying that native should not require an extra macro.  And also WITHOUT_PTHREADS is misleading if it only turns off pthreads in one place.  

We use the following to identify WASM:
__EMSCRIPTEN__
WASM

and you may as well use one of those.

Reply:
Using `__EMSCRIPTEN__`.